eventdemo/source/c/src/kaa_demo.c

```diff
@@ -240,14 +107,184 @@ kaa_error_t kaa_on_attach_success(void *context)
 
 kaa_error_t kaa_on_attach_failed(void *context, user_verifier_error_code_t error_code, const char *reason)
 {
-    KAA_DEMO_UNUSED(context);
+    (void)context;
 
     printf("Kaa Demo attach failed: error %d, reason '%s'\n", error_code, (reason ? reason : "null"));
 
-    is_shutdown = true;
     return KAA_ERR_NONE;
 }
 
+void kaa_on_change_chat(void *context, kaa_chat_chat_event_t *event, kaa_endpoint_id_p source)
+{
+    (void)context;
+    (void)source;
+
+    if (event->event_type == ENUM_CHAT_EVENT_TYPE_CREATE) {
+        pthread_mutex_lock(&lock);
+        kaa_list_push_back(lst, event->chat_name->data);
+        pthread_mutex_unlock(&lock);
+    } else {
+        pthread_mutex_lock(&lock);
+        kaa_list_remove_first(kaa_list_begin(&lst), &rooms_equal, event->chat_name->data, NULL);
+        pthread_mutex_unlock(&lock);
+    }
+}
+
+void kaa_on_receive_message(void *context, kaa_chat_message_t *event, kaa_endpoint_id_p source)
+{
+    if (!strncmp(current_room, event->chat_name->data, strnlen(current_room, KAA_STRING_MAX_LENGTH))) {
+        puts(event->message->data);
```

Comment:
Is it guaranteed the event->message->data is NUL-ed to be a C-string?
Otherwise it would be better to use current_room here as they match within the checked range but can guarantee to be C-string.